pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -3840,6 +3836,15 @@ func TestResolvePipelineRunTask_WithMatrix(t *testing.T) {
 			PipelineTask: &pts[1],
 			ResolvedTask: rtr,
 		},
+	}, {
+		name: "task with matrix - whole array results",
+		pt:   pts[2],
+		want: &ResolvedPipelineTask{
+			TaskRunNames: nil,
+			TaskRuns:     nil,
+			PipelineTask: &pts[2],
+			ResolvedTask: nil,
```

Comment:
This case will actually get error since we pass `nil` as the `pst`, but we don't check the error there. I think if we choose the current approach we need to add pst for this test.

Reply:
Added passing in pst for the whole array results.